src/libsync/propagatedownload.cpp

```diff
@@ -601,7 +657,7 @@ void PropagateDownloadFile::slotGetFinished()
         return;
     }
 
-    if (bodySize > 0 && bodySize != _tmpFile.size() - job->resumeStart()) {
+    if (!_isDeltaSyncDownload && bodySize > 0 && bodySize != _tmpFile.size() - job->resumeStart()) {
```

Comment:
Do you have similar check for the delta sync?

This test exist because sometimes, the GET might end without error, but only contains a turncated file. (this happens sometimes when there is a gateway and the http chunk encoding is used)

Maybe if there is  checksum check and that the file is correct, or the zsync algo detect corrupted patch files?

Reply:
The final checksum check would definitely catch this.